codegen-server/typescript/src/main/kotlin/software/amazon/smithy/rust/codegen/server/typescript/smithy/generators/TsApplicationGenerator.kt

```diff
@@ -0,0 +1,303 @@
+            pub struct TsSocket(#{socket2}::Socket);
+            """,
+            *codegenScope,
+        )
+
+        Attribute("napi").render(writer)
+        writer.rustBlock("impl TsSocket") {
+            writer.rust(
+                """
+                /// Create a new UNIX `Socket` from an address, port and backlog.
+                /// If not specified, the backlog defaults to 1024 connections.
+                """.trimIndent(),
+            )
+            Attribute("napi(constructor)").render(writer)
+            writer.rustBlockTemplate(
+                """pub fn new(address: String, port: i32, backlog: Option<i32>) -> #{napi}::Result<Self>""".trimIndent(),
+                *codegenScope,
+            ) {
+                writer.rustTemplate(
+                    // XXX How do I make the new_socket import here?
+                    // In the previous file, it was use aws_smithy_http_server::socket::new_socket;
+                    """
+                    Ok(Self(
+                        aws_smithy_http_server::socket::new_socket(address, port, backlog)
+                        .map_err(|e| #{napi}::Error::from_reason(e.to_string()))?,
+                    ))
+                    """.trimIndent(),
+                    *codegenScope,
+                )
+            }
+
+            writer.rust(
+                """
+                /// Clone the inner socket allowing it to be shared between multiple
+                /// Nodejs processes.
+                """.trimIndent(),
+            )
+            Attribute("napi").render(writer)
+            writer.rustBlockTemplate(
+                // XXX Can I change the signature to #{napi}::Result<Self>?
+                """pub fn try_clone(&self) -> #{napi}::Result<TsSocket>""".trimIndent(),
+                *codegenScope,
+            ) {
+                writer.rustTemplate(
+                    """
+                    Ok(TsSocket(
+                        self.0
+                            .try_clone()
+                            .map_err(|e| #{napi}::Error::from_reason(e.to_string()))?,
+                    ))
+                    """.trimIndent(),
+                    *codegenScope,
+                )
+            }
+        }
+
+        writer.rustBlock("impl TsSocket") {
+            writer.rustBlockTemplate(
+                """pub fn to_raw_socket(&self) -> #{napi}::Result<#{socket2}::Socket>""".trimIndent(),
+                *codegenScope,
+            ) {
+                writer.rustTemplate(
+                    """
+                    self.0
+                     .try_clone()
+                     .map_err(|e| #{napi}::Error::from_reason(e.to_string()))
+                    """.trimIndent(),
+                    *codegenScope,
+                )
+            }
+        }
+    }
+
+    private fun renderServer(writer: RustWriter) {
+        writer.rustBlockTemplate(
+            // / XXX Check here the aws_smithy_http_server (same issue with new_socket above)
```

Comment:
Same, you just need to use reference `#{SmithyServer}` instead of `aws_smithy_http_server`